File: lista/Python/questao2.py
"""
 Adicione um método acelerar e frear à classe Carro que altere um atributo
velocidade. Crie um método para exibir a velocidade atual.

"""
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, potencia):
        print("Acelerando...\n")
        self.velocidade += potencia

    def frear(self):
        print("Freando...\n")
        self.velocidade = self.velocidade // 2

    def exibir_carro(self):
        print(f"Marca: {self.marca}")
        print(f"Modelo: {self.modelo}")
        print(f"Ano: {self.ano}")
        print()

    def exibir_velocidade(self):
        print(f"Velocidade Atual: {self.velocidade}\n")

def novo_carro(marca, modelo, ano):
    return Carro(marca, modelo, ano)

def main():
    carro1 = novo_carro("Fiat", "Uno", 2020)

    carro1.exibir_carro()

    carro1.acelerar(10)
    carro1.exibir_velocidade()
    carro1.acelerar(30)
    carro1.exibir_velocidade()

    carro1.frear()
    carro1.frear()
    carro1.frear()
    carro1.exibir_velocidade()

if __name__ == "__main__":
    main()

File: lista/GO/questao2/questao2.go
package main

import (
	"fmt"
)

// Definindo a estrutura Carro
type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
}

// Função para criar um novo Carro
func NovoCarro(marca, modelo string, ano int) Carro {
	return Carro{Marca: marca, Modelo: modelo, Ano: ano, Velocidade: 0}
}

// Método para acelerar o carro
func (c *Carro) Acelerar(potencia int) {
	fmt.Println("Acelerando...\n")
	c.Velocidade += potencia
}

// Método para frear o carro
func (c *Carro) Frear() {
	fmt.Println("Freando...\n")
	c.Velocidade = c.Velocidade / 2
}

// Método para exibir detalhes do carro
func (c Carro) ExibirCarro() {
	fmt.Printf("Marca: %s\n", c.Marca)
	fmt.Printf("Modelo: %s\n", c.Modelo)
	fmt.Printf("Ano: %d\n", c.Ano)
	fmt.Println()
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("Velocidade Atual: %d\n\n", c.Velocidade)
}

func main() {
	carro1 := NovoCarro("Fiat", "Uno", 2020)

	carro1.ExibirCarro()

	carro1.Acelerar(10)
	carro1.ExibirVelocidade()
	carro1.Acelerar(30)
	carro1.ExibirVelocidade()

	carro1.Frear()
	carro1.Frear()
	carro1.Frear()
	carro1.ExibirVelocidade()
}
